pkg/minishift/docker/image/oci_image_handler.go

```diff
@@ -331,21 +334,54 @@ func (handler *OciImageHandler) exportImage(image string, config *ImageCacheConf
 		return fmt.Errorf("Invalid image source '%s': %v", image, err)
 	}
 
-	destRef, err := layout.NewReference(config.HostCacheDir, image)
+	// tmpImageIndexLocation should be a directory location which will be atomic for each image.
+	// for an image "openshift/origin-control-plane:v3.10.0"
+	// it will be $HOME/.minishift/cache/image/openshift-origin-control-plane-v3.10.0
+	r := strings.NewReplacer(":", "-", "/", "-")
+	tmpImageIndexLocation := filepath.Join(config.HostCacheDir, r.Replace(image))
```

Comment:
Temporary directory name should not be manually created. It should be auto generated which comes with random number etc. Also we have a location i.e. `~/.minishift/tmp` , so we should use that location for temporary directory/file. 

If you run the same image export command from two shells it will have same directory name, which is not desirable.

Reply:
@LalatenduMohanty No so what we need to do is stop the that operation, if a user try to export same image in 2 different terminal we shouldn't allow it. I will make that change but using the `tmp` module is not what I am after atm.